nuget/helpers/build

```diff
@@ -32,7 +32,7 @@ cd "$install_dir/lib/NuGetUpdater/NuGetUpdater.Cli"
 dotnet publish \
     --configuration Release \
     --output "$install_dir/NuGetUpdater" \
-    --framework net8.0 \
+    --framework net9.0 \
```

Comment:
To test `net9.0`, the tests have to target `net9.0` due to this logic for setting up the local package repository used in the tests:

https://github.com/dependabot/dependabot-core/blob/fa8d392d2afc5cdbdcea620b5413d023afddff50/nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/MockNuGetPackage.cs#L332-L337

Otherwise the file read doesn't contain support for .NET 9, and then the local package repo doesn't contain any of the required fake packages and `dotnet restore` fails for `net9.0` projects.